Tests/UnitTests/EdgeHitProcessorTests.swift

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPCore
+@testable import AEPEdge
+import AEPServices
+import XCTest
+
+class EdgeHitProcessorTests: XCTestCase {
+    var hitProcessor: EdgeHitProcessor!
+    var networkService: EdgeNetworkService!
+    var networkResponseHandler: NetworkResponseHandler!
+    var mockNetworkService: MockNetworking? {
+        return ServiceProvider.shared.networkService as? MockNetworking
+    }
+
+    override func setUp() {
+        ServiceProvider.shared.networkService = MockNetworking()
+        networkService = EdgeNetworkService()
+        networkResponseHandler = NetworkResponseHandler()
+        hitProcessor = EdgeHitProcessor(networkService: networkService, networkResponseHandler: networkResponseHandler)
+    }
+
+    /// Tests that when a `DataEntity` with bad data is passed, that it is not retried and is removed from the queue
+    func testProcessHitBadHit() {
+        // setup
+        let expectation = XCTestExpectation(description: "Callback should be invoked with true signaling this hit should not be retried")
+        let entity = DataEntity(uniqueIdentifier: "test-uuid", timestamp: Date(), data: nil) // entity data does not contain an `EdgeHit`
+
+        // test
+        hitProcessor.processHit(entity: entity) { success in
+            XCTAssertTrue(success)
+            expectation.fulfill()
+        }
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse(mockNetworkService?.connectAsyncCalled ?? true) // no network request should have been made
+    }
+
+    /// Tests that when a good hit is processed that a network request is made and the request returns 200
+    func testProcessHitHappy() {
```

Comment:
testProcessHit_happy_sendsNetworkRequest_returnsTrue

Reply:
Hmmm, we're using a different naming convention in our tests in Swift, but I'll update these to match edge.